### PR TITLE
Adding the flage --imagepullpolicy and implemented on the jobs to pull the image

### PR DESCRIPTION
**Pull request by kannanvj**
Adding the flage --imagepullpolicy and implemented on the jobs to pull the image

### CODE REVIEW

File: cmd/app/controller.go [hunk 1]
```diff
@@ -92,7 +92,8 @@ func NewController(
 	imageCacheInformer informers.ImageCacheInformer,
 	imageCacheRefreshFrequency time.Duration,
 	imagePullDeadlineDuration time.Duration,
-	dockerClientImage string) *Controller {
+	dockerClientImage string,
+	imagepullpolicy string) *Controller {
```
Comment (senthilrch) [inline cmd/app/controller.go]:
```suggestion
	imagePullPolicy string) *Controller {
```

File: cmd/app/controller.go [hunk 2]
```diff
@@ -114,7 +115,7 @@ func NewController(
 		imageCacheRefreshFrequency: imageCacheRefreshFrequency,
 	}
 
-	imageManager := images.NewImageManager(controller.workqueue, controller.imageworkqueue, controller.kubeclientset, fledgedNameSpace, imagePullDeadlineDuration, dockerClientImage)
+	imageManager := images.NewImageManager(controller.workqueue, controller.imageworkqueue, controller.kubeclientset, fledgedNameSpace, imagePullDeadlineDuration, dockerClientImage, imagepullpolicy)
```
Comment (senthilrch) [inline cmd/app/controller.go]:
```suggestion
	imageManager := images.NewImageManager(controller.workqueue, controller.imageworkqueue, controller.kubeclientset, fledgedNameSpace, imagePullDeadlineDuration, dockerClientImage, imagePullPolicy)
```

File: cmd/fledged.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ var (
 	imageCacheRefreshFrequency time.Duration
 	imagePullDeadlineDuration  time.Duration
 	dockerClientImage          string
+	imagepullpolicy            string
```
Comment (senthilrch) [inline cmd/fledged.go]:
```suggestion
	imagePullPolicy            string
```

File: cmd/fledged.go [hunk 4]
```diff
@@ -67,7 +68,7 @@ func main() {
 	controller := app.NewController(kubeClient, fledgedClient,
 		kubeInformerFactory.Core().V1().Nodes(),
 		fledgedInformerFactory.Fledged().V1alpha1().ImageCaches(),
-		imageCacheRefreshFrequency, imagePullDeadlineDuration, dockerClientImage)
+		imageCacheRefreshFrequency, imagePullDeadlineDuration, dockerClientImage, imagepullpolicy)
```
Comment (senthilrch) [inline cmd/fledged.go]:
```suggestion
		imageCacheRefreshFrequency, imagePullDeadlineDuration, dockerClientImage, imagePullPolicy)
```

File: cmd/fledged.go [hunk 5]
```diff
@@ -89,4 +90,5 @@ func init() {
 	flag.DurationVar(&imagePullDeadlineDuration, "image-pull-deadline-duration", time.Minute*5, "Maximum duration allowed for pulling an image. After this duration, image pull is considered to have failed")
 	flag.DurationVar(&imageCacheRefreshFrequency, "image-cache-refresh-frequency", time.Minute*15, "The image cache is refreshed periodically to ensure the cache is up to date. Setting this flag to 0s will disable refresh")
 	flag.StringVar(&dockerClientImage, "docker-client-image", "senthilrch/fledged-docker-client:latest", "The image name of the docker client. the docker client is used when deleting images during purging the cache")
+	flag.StringVar(&imagepullpolicy, "imagepullpolicy", "", "Image pull policy default value is IfNotPresent. Possible values are IfNotPresent and Always")
```
Comment (senthilrch) [inline cmd/fledged.go]:
```suggestion
	flag.StringVar(&imagePullPolicy, "image-pull-policy", "", "Image pull policy for pulling images into the cache. Default value is 'IfNotPresent'. Possible values are 'IfNotPresent' and 'Always'")
```

File: deploy/fledged-deployment.yaml [hunk 6]
```diff
@@ -26,6 +26,7 @@ spec:
         - "--image-pull-deadline-duration=5m"
         - "--image-cache-refresh-frequency=0s"
         - "--docker-client-image=senthilrch/fledged-docker-client:v0.1.3"
+        - "--imagepullpolicy=IfNotPresent"
```
Comment (senthilrch) [inline deploy/fledged-deployment.yaml]:
```suggestion
        - "--image-pull-policy=IfNotPresent"
```

File: pkg/images/image_helpers.go [hunk 7]
```diff
@@ -29,11 +30,20 @@ import (
 )
 
 // newImagePullJob constructs a job manifest for pulling an image to a node
-func newImagePullJob(imagecache *fledgedv1alpha1.ImageCache, image string, hostname string) (*batchv1.Job, error) {
+func newImagePullJob(imagecache *fledgedv1alpha1.ImageCache, image string, hostname string, imagepullpolicy string) (*batchv1.Job, error) {
```
Comment (senthilrch) [inline pkg/images/image_helpers.go]:
```suggestion
func newImagePullJob(imagecache *fledgedv1alpha1.ImageCache, image string, hostname string, imagePullPolicy string) (*batchv1.Job, error) {
```

File: pkg/images/image_helpers.go [hunk 8]
```diff
@@ -29,11 +30,20 @@ import (
...
+	var pullPolicy corev1.PullPolicy = corev1.PullIfNotPresent
+
 	if imagecache == nil {
 		glog.Error("imagecache pointer is nil")
 		return nil, fmt.Errorf("imagecache pointer is nil")
 	}
+	if imagepullpolicy == string(corev1.PullAlways) {
```
Comment (senthilrch) [inline pkg/images/image_helpers.go]:
```suggestion
	if imagePullPolicy == string(corev1.PullAlways) {
```

File: pkg/images/image_helpers.go [hunk 9]
```diff
@@ -29,11 +30,20 @@ import (
...
+		pullPolicy = corev1.PullAlways
+	} else if imagepullpolicy == "" {
```
Comment (senthilrch) [inline pkg/images/image_helpers.go]:
```suggestion
	} else if imagePullPolicy == "" {
```

File: pkg/images/image_manager.go [hunk 10]
```diff
@@ -59,6 +59,7 @@ type ImageManager struct {
 	podsSynced                cache.InformerSynced
 	imagePullDeadlineDuration time.Duration
 	dockerClientImage         string
+	imagepullpolicy           string
```
Comment (senthilrch) [inline pkg/images/image_manager.go]:
```suggestion
	imagePullPolicy           string
```

File: pkg/images/image_manager.go [hunk 11]
```diff
@@ -105,7 +106,7 @@ func NewImageManager(
 	kubeclientset kubernetes.Interface,
 	namespace string,
 	imagePullDeadlineDuration time.Duration,
-	dockerClientImage string) *ImageManager {
+	dockerClientImage, imagepullpolicy string) *ImageManager {
```
Comment (senthilrch) [inline pkg/images/image_manager.go]:
```suggestion
	dockerClientImage, imagePullPolicy string) *ImageManager {
```

File: pkg/images/image_manager.go [hunk 12]
```diff
@@ -123,6 +124,7 @@ func NewImageManager(
 		podsSynced:                podInformer.Informer().HasSynced,
 		imagePullDeadlineDuration: imagePullDeadlineDuration,
 		dockerClientImage:         dockerClientImage,
+		imagepullpolicy:           imagepullpolicy,
```
Comment (senthilrch) [inline pkg/images/image_manager.go]:
```suggestion
		imagePullPolicy:           imagePullPolicy,
```

File: pkg/images/image_manager.go [hunk 13]
```diff
@@ -397,7 +399,7 @@ func (m *ImageManager) processNextWorkItem() bool {
 // pullImage pulls the image to the node
 func (m *ImageManager) pullImage(iwr ImageWorkRequest) (*batchv1.Job, error) {
 	// Construct the Job manifest
-	newjob, err := newImagePullJob(iwr.Imagecache, iwr.Image, iwr.Node)
+	newjob, err := newImagePullJob(iwr.Imagecache, iwr.Image, iwr.Node, m.imagepullpolicy)
```
Comment (senthilrch) [inline pkg/images/image_manager.go]:
```suggestion
	newjob, err := newImagePullJob(iwr.Imagecache, iwr.Image, iwr.Node, m.imagePullPolicy)
```

File: cmd/app/controller.go (hunk 2 above)
Comment (kannanvj) [reply cmd/app/controller.go]: Done the changes

**Pull request merged**